docs/tradeoffs.md

```diff
@@ -0,0 +1,135 @@
+# Design Trade-offs
+
+This will be looking at CLI parser design trade offs from the lens of comparing
+[bpaf](https://docs.rs/bpaf) and [clap](https://docs.rs/clap).
+
+For anyone asking the question "which should I use?", the short answer would be
+- `bpaf` would work well for simple cases like example bins for a library
+  while giving the best build times and a reasonable to understand code with
+  the `derive` API
+- All other cases the answer is "it depends" as the two designs offer different
+  trade offs.   If you want to just choose one with little research that will
+  cover the most use cases, that will most likely be `clap`.
```

Comment:
Overall, we're trying to move more towards the middle, to be user friendly in the common cases but to provide building blocks for the less common cases so that (1) there is less overload in the documentation, (2) we compile faster, and (3) people can build the solutions they need without getting every feature baked directly in to clap.

Through v3 we had done some of the initial work in that direction which will be more obvious when I release v4 (wrapping up the documentation for the release).

Reply:
Okay, similar goal, approaching from different directions :+1: